compiler/rustc_middle/src/ty/query.rs

```diff
@@ -105,6 +110,15 @@ impl<'tcx> TyCtxt<'tcx> {
         TyCtxtEnsure { tcx: self }
     }
 
+    /// Returns a transparent wrapper for `TyCtxt`, which ensures queries
+    /// are executed instead of just returning their results.
+    ///
+    /// This version verifies that the computed result exists in the cache before returning.
+    #[inline(always)]
+    pub fn ensure_with_value(self) -> TyCtxtEnsureWithValue<'tcx> {
```

Comment:
What do you think about flipping the defaults just to nudge people in the right direction? So renaming `ensure` to `ensure_without_checking_incremental_cache` and renaming `ensure_with_value` to `ensure`.

Reply:
There are currently 126 uses of `ensure()` in the codebase, and only <13 have a link with stealing a query result.
Most of them are in rustc_hir_analysis and rustc_interface as ways to drive analysis.

I can rename both to `ensure_unchanged` (current behaviour, to drive analyses) and `ensure_with_value` (new behaviour, for `steal`) to make everything explicit.

Reply:
oof that's gonna be very verbose...

let's land this and I'll think about it a bit more